### PR TITLE
Paladin namespace is scoped for being instanced, but still global when possible

### PR DESCRIPTION
**Pull request by secretrobotron**
1. Subsystems and Paladin core objects enclosed in instance of Paladin object and automatically initialized
2. Paladin namespace is overwritten when new Paladin() occurs
3. Tests updated to use paladin instead of Paladin (and no Game wrapper)
4. Subsystems altered to return an instantiable object, which is constructed when Paladin.subsystem.init occurs

### CODE REVIEW

File: src/paladin.js [hunk 1]
```diff
@@ -6,646 +6,654 @@ Array.prototype.remove = function(from, to) {
...
-};
-Component.prototype.getType = function() {
-    return this.type;
-};
-Component.prototype.getSubtype = function() {
-    return this.subtype;
-};
-
-function SpatialComponent( position, rotation ) {
+var Paladin = window.Paladin = function ( options ) {
```
Comment (modeswitch) [inline src/paladin.js:16]: I don't think we want to do this. It makes it harder to keep instances separated. If the client code wishes to do this then that's fine, but we shouldn't do it for them.

**Comment by modeswitch** [comment]
Since we want to allow multiple simultaneous instances of paladin, having any global state is going to be messy. So I think "still global when possible" is not something we want to do here.

File: src/paladin.js (hunk 1 above)
Comment (secretrobotron) [inline src/paladin.js:16]: What in particular? window.Paladin?

The "var Paladin" part is just for the scope in this file (convenience).

**Comment by secretrobotron** [comment]
I'm glad you agree with clearing out the global scope. I left it in for legacy (just in case), but editing the tests turned out to be tremendously simple without it.

Consider it done.

File: src/paladin.js (hunk 1 above)
Comment (modeswitch) [inline src/paladin.js:16]: Yes, window.Paladin.

Comment (modeswitch) [inline src/paladin.js:16]: Oops, nm. Comment withdrawn.

**Comment by secretrobotron** [comment]
Fixed up a few things and got rid of globally-scoped Paladin objects. You can do everything with a Paladin instance now (and, optionally, within its setup function).

**Pull request merged**